rs/nervous_system/candid_utils/src/printing/tests.rs

```diff
@@ -0,0 +1,78 @@
+use super::*;
+
+#[derive(candid::CandidType, Clone, PartialEq)]
+struct DummyCandidStruct {
+    pub status: Option<i32>,
+    pub module_hash: Vec<u8>,
+    pub controllers: String,
+    pub memory_size: Option<u64>,
+    pub cycles: Option<u64>,
+}
+
+#[derive(candid::CandidType, Clone, PartialEq)]
+enum DummyCandidVariant {
+    Foo(String),
+    Bar { abc: String, xyz: String },
+}
+
+#[derive(candid::CandidType, Clone, PartialEq)]
+struct DummyCandidContainer {
+    foo: DummyCandidVariant,
+    bar: Result<DummyCandidVariant, String>,
+}
+
+#[track_caller]
```

Comment:
I did not know about this feature! I will definitely be using this in many of my future tests! This is going to make my tests' fail output more useful 🥰